internal/controllers/extension_controller.go

```diff
@@ -156,10 +227,161 @@ func (r *ExtensionReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&ocv1alpha1.Extension{}).
+		Owns(&kappctrlv1alpha1.App{}).
 		Watches(&catalogd.Catalog{}, handler.EnqueueRequestsFromMapFunc(extensionRequestsForCatalog(mgr.GetClient(), mgr.GetLogger()))).
 		Complete(r)
 }
 
+// TODO: follow up with mapping of all the available App statuses: https://github.com/carvel-dev/kapp-controller/blob/855063edee53315811a13ee8d5df1431ba258ede/pkg/apis/kappctrl/v1alpha1/status.go#L28-L35
+// mapAppStatusToInstalledCondition currently maps only the installed condition.
+func mapAppStatusToInstalledCondition(existingApp *kappctrlv1alpha1.App, ext *ocv1alpha1.Extension, bundleImage string) {
```

Comment:
We may want another `Extension` condition type as well. I could see a `Progressing` type that is populated based on App conditions when the App is not `ReconcileSucceeded`.

This could be part of the followup you mention in the TODO comment.

Reply:
Yes, this is a part of #655, we need additional condition types to track - Reconciling, ReconcileFailed, Deleting, DeletingFailed statuses as provided in the App. Created a separate issue to discuss more about them.